src/SymWoodburyMatrices.jl

```diff
@@ -0,0 +1,160 @@
+  calc_inv(O.A, O.B, full(O.D));
+
+\(W::SymWoodbury, R::StridedVecOrMat) = inv(W)*R
+
+"""
+    partialInv(O)
+
+Get the factors (X,Z) in W + XZXᵀ where W + XZXᵀ = inv( A + BDBᵀ )
+"""
+function partialInv(O::SymWoodbury)
+
+  X = (O.A)\O.B;
+  invD = -1*inv(O.D);
+  Z = inv(invD - O.B'*X);
+  return (X,Z);
+
+end
+
+function liftFactorVars(A,B,Di)
+  n  = size(A,1)
+  k  = size(B,2)
+  M = [A    B   ;
+       B'  -Di ];
+  M = lufact(M) # ldltfact once it's available.
+  return x -> (M\[x; zeros(k,1)])[1:n,:];
+end
+
+"""
+    liftFactor(A)
+
+More stable version of inv(A).  Returns a function which computs the inverse
+on evaluation, i.e. `liftFactor(A)(x)` is the same as `inv(A)*x`.
+"""
+liftFactor{T<:Any,
+           AType<:Any, 
+           BType<:Any, 
+           DType<:Matrix}(O::SymWoodbury{T,AType,BType,DType}) = 
+           liftFactorVars(sparse(O.A), sparse(O.B), sparse(inv(O.D))); 
+
+liftFactor{T<:Any,
+           AType<:Any, 
+           BType<:Any, 
+           DType<:SparseMatrixCSC}(O::SymWoodbury{T,AType,BType,DType}) = 
+           liftFactorVars(sparse(O.A), sparse(O.B), sparse(inv(full(O.D))));
+
+# Optimization - use specialized BLAS package 
+function *{T<:Any, AType<:Any, BType<:Matrix, DType<:Any}(O::SymWoodbury{T,AType,BType,DType}, 
+  x::Array{Float64,2})
+  o = O.A*x;
+  w = O.D*gemm('T','N',O.B,x);
+  gemm!('N','N',1.,O.B,w,1., o)
+  return o
+end
+
+function *{T<:Any, AType<:Any, BType<:Any, DType<:Any}(O::SymWoodbury{T,AType,BType,DType}, 
+  x::VectorTypes)
+  return O.A*x + O.B*(O.D*(O.B'x));
+end
+
+Base.Ac_mul_B(O1::SymWoodbury, x::VectorTypes) = O1*x
+
+
++(O::SymWoodbury, M::SymWoodbury)    = SymWoodbury(O.A + M.A, [O.B M.B],
+                                                   cat([1,2],O.D,M.D) );
+*(α::Real, O::SymWoodbury)           = SymWoodbury(α*O.A, O.B, α*O.D);
+*(O::SymWoodbury, α::Real)           = SymWoodbury(α*O.A, O.B, α*O.D);
++(M::AbstractMatrix, O::SymWoodbury) = SymWoodbury(O.A + M, O.B, O.D);
++(O::SymWoodbury, M::AbstractMatrix) = SymWoodbury(O.A + M, O.B, O.D);
+Base.size(M::SymWoodbury)            = size(M.A);
+Base.size(M::SymWoodbury, i)         = (i == 1 || i == 2) ? size(M)[1] : 1
+
+Base.full{T}(O::SymWoodbury{T})      = full(O.A) + O.B*O.D*O.B'
+Base.copy{T}(O::SymWoodbury{T})      = SymWoodbury(copy(O.A), copy(O.B), copy(O.D))
+
+function square(O::SymWoodbury)
+  A = O.A^2
+  D = O.D
+  B = O.B
+  Z = [(O.A*B + B) (O.A*B - B) B]
+  SymWoodbury(A, Z, full( cat([1,2],D/2,-D/2, D*B'*B*D) ) )
+end
+
+"""
+The product of two SymWoodbury matrices will generally be a Woodbury Matrix,
+except when they are the same, i.e. the user writes A'A or A*A' or A*A.
+
+Z(A + B*D*Bᵀ) = ZA + ZB*D*Bᵀ
+"""
+function *(O1::SymWoodbury, O2::SymWoodbury)
+  if (O1 === O2) 
+    return square(O1)
+  else
+    if O1.A == O2.A && O1.B == O2.B && O1.D == O2.D
+      return square(O1)
+    else
+      throw(DomainError())
```

Comment:
Just checking, you're explicitly not supporting the general product of two `SymWoodbury` matrices, right?  Instead of a `DomainError` (which is intended for single numbers in high-performance situations, not relevant here), how about throwing an `ArgumentError` where you suggest converting to `Woodbury`? (Looks like you haven't defined a `convert{W<:Woodbury}(::Type{W}, O::SymWoodbury)` method yet.)